packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -0,0 +1,1599 @@
+  bool get cutEnabled => false;
+
+  @override
+  bool get pasteEnabled => false;
+
+  @override
+  void hideToolbar([bool hideHandles = true]) {
+    _selectionOverlay?.hideToolbar();
+    if (hideHandles) {
+      _selectionOverlay?.hideToolbar();
+    }
+  }
+
+  @override
+  void selectAll([SelectionChangedCause? cause]) {
+    _clearSelection();
+    _selectable?.dispatchSelectionEvent(const SelectAllSelectionEvent());
+    if (cause == SelectionChangedCause.toolbar) {
+      _showToolbar();
+      _showHandles();
+    }
+  }
+
+  @override
+  void copySelection(SelectionChangedCause cause) {
+    _copy();
+    _clearSelection();
+  }
+
+  // TODO(chunhtai): remove this workaround after decoupling text selection
+  // from text editing in TextSelectionDelegate.
+  @override
+  TextEditingValue textEditingValue = const TextEditingValue(text: '_');
+
+  @override
+  void bringIntoView(TextPosition position) {/* SelectableRegion must be in view at this point. */}
+
+  @override
+  void cutSelection(SelectionChangedCause cause) {
+    assert(false);
+  }
+
+  @override
+  void userUpdateTextEditingValue(TextEditingValue value, SelectionChangedCause cause) {/* SelectableRegion maintains its own state */}
+
+  @override
+  Future<void> pasteText(SelectionChangedCause cause) async {
+    assert(false);
+  }
+
+  // [SelectionRegistrar] override.
+
+  @override
+  void add(Selectable selectable) {
+    assert(_selectable == null);
+    _selectable = selectable;
+    _selectable!.addListener(_updateSelectionStatus);
+    _selectable!.pushHandleLayers(_startHandleLayerLink, _endHandleLayerLink);
+  }
+
+  @override
+  void remove(Selectable selectable) {
+    assert(_selectable == selectable);
+    _selectable!.removeListener(_updateSelectionStatus);
+    _selectable!.pushHandleLayers(null, null);
+    _selectable = null;
+  }
+
+  @override
+  void dispose() {
+    _selectable?.removeListener(_updateSelectionStatus);
+    _selectable?.pushHandleLayers(null, null);
+    _selectionDelegate.dispose();
+    _selectionOverlay?.dispose();
+    _selectionOverlay = null;
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CompositedTransformTarget(
+      link: _toolbarLayerLink,
+      child: RawGestureDetector(
+        gestures: _gestureRecognizers,
+        behavior: HitTestBehavior.translucent,
+        excludeFromSemantics: true,
+        child: Actions(
+          actions: _actions,
+          child: Focus(
+            focusNode: widget.focusNode,
+            child: SelectionContainer(
+              registrar: this,
+              delegate: _selectionDelegate,
+              child: widget.child,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/// An action that does not override any [Action.overridable] in the subtree.
+///
+/// If a intent is fire at or below the context of [Action.overridable] in the
+/// subtree, this action will respect the original action and not override it.
```

Comment:
What would happen if someone makes _NonOverrideAction overridable? :exploding_head: Is it worth documenting that case?

Reply:
selectable region already does it. so unfortunately, the user can't tell whether they are override the SelectableRegion or EditableText's actions.

but if they also use _NonOverrideAction, every thing will work as expected. Intent fired from from EditableText's it will fire bubble up to the top most _NonOverrideAction which will call a chain of callingActions back to the EditableText. Intent fired from SelectableRegion, also bubble up to top most _NonOverrideAction and chain of callingAction will only go back to SelectableRegion because SelectableRegion's callingAction will be null